src/chunk.h

```diff
@@ -101,31 +101,31 @@ typedef enum CascadeToMaterializationOption
 extern Chunk *ts_chunk_create(Hypertable *ht, Point *p, const char *schema, const char *prefix);
 extern TSDLLEXPORT Chunk *ts_chunk_create_base(int32 id, int16 num_constraints);
 extern TSDLLEXPORT ChunkStub *ts_chunk_stub_create(int32 id, int16 num_constraints);
-extern Chunk *ts_chunk_find(Hyperspace *hs, Point *p, bool include_chunks_marked_as_dropped);
+extern Chunk *ts_chunk_find(Hypertable *ht, Point *p);
 extern Chunk **ts_chunk_find_all(Hyperspace *hs, List *dimension_vecs, LOCKMODE lockmode,
 								 unsigned int *num_chunks);
 extern List *ts_chunk_find_all_oids(Hyperspace *hs, List *dimension_vecs, LOCKMODE lockmode);
 extern TSDLLEXPORT int ts_chunk_add_constraints(Chunk *chunk);
 
 extern Chunk *ts_chunk_copy(Chunk *chunk);
 extern Chunk *ts_chunk_get_by_name_with_memory_context(const char *schema_name,
-													   const char *table_name,
-													   int16 num_constraints, MemoryContext mctx,
+													   const char *table_name, MemoryContext mctx,
 													   bool fail_if_not_found);
 extern TSDLLEXPORT void ts_chunk_insert_lock(Chunk *chunk, LOCKMODE lock);
 
-extern TSDLLEXPORT Oid ts_chunk_create_table(Chunk *chunk, Hypertable *ht, char *tablespacename);
-extern TSDLLEXPORT Chunk *ts_chunk_get_by_id(int32 id, int16 num_constraints,
-											 bool fail_if_not_found);
-extern TSDLLEXPORT Chunk *ts_chunk_get_by_relid(Oid relid, int16 num_constraints,
-												bool fail_if_not_found);
+extern TSDLLEXPORT Oid ts_chunk_create_table(Chunk *chunk, Hypertable *ht,
+											 const char *tablespacename);
+extern TSDLLEXPORT Chunk *ts_chunk_get_by_id(int32 id, bool fail_if_not_found);
+extern TSDLLEXPORT Chunk *ts_chunk_get_by_relid(Oid relid, bool fail_if_not_found);
 extern bool ts_chunk_exists(const char *schema_name, const char *table_name);
+extern Oid ts_chunk_get_relid(int32 chunk_id, bool missing_ok);
+extern Oid ts_chunk_get_schema_id(int32 chunk_id, bool missing_ok);
 extern bool ts_chunk_exists_relid(Oid relid);
 
 extern TSDLLEXPORT bool ts_chunk_exists_with_compression(int32 hypertable_id);
 extern void ts_chunk_recreate_all_constraints_for_dimension(Hyperspace *hs, int32 dimension_id);
-extern TSDLLEXPORT void ts_chunk_drop_fks(Chunk *chunk);
-extern TSDLLEXPORT void ts_chunk_create_fks(Chunk *chunk);
+extern TSDLLEXPORT void ts_chunk_drop_fks(Chunk *const chunk);
```

Comment:
You expose an implementation detail, which only affects the usage of the variable inside the function implementation, and has no sense for the function user. I suggest to remove this and below changes.

Reply:
The microsoft compiler warns against having different declarations, so if we are to enable warnings as errors, we need to keep declaration and definition consistent. I also think that it is good rule in general to keep consistency.

Reply:
@erimatnor Good to know. What version do you use? Is it through command line or Visual Studio?

Reply:
I believe I used Visual Studio 2017, both via the command-line and Visual Studio.

Reply:
I haven't noticed these warnings in 2019, but I didn't pay too much attention. I will double check how it works there.